src/mono/wasm/runtime/method-calls.ts

```diff
@@ -301,17 +301,21 @@ export function mono_call_assembly_entry_point(assembly: string, args?: any[], s
     return mono_bind_assembly_entry_point(assembly, signature)(...args);
 }
 
-export function mono_wasm_invoke_js_with_args(js_handle: JSHandle, method_name: MonoString, args: MonoArray, is_exception: Int32Ptr): any {
-    const argsRoot = mono_wasm_new_root(args), nameRoot = mono_wasm_new_root(method_name);
+export function mono_wasm_invoke_js_with_args(js_handle: JSHandle, method_name: MonoString, args: MonoArray, is_exception: Int32Ptr, result_address: MonoObjectRef): any {
```

Comment:
method_name should become MonoStringRef, which will turn nameRoot into an external root. args also needs to become MonoObjectRef and argsRoot becomes external root as well